js/app/widgets/eknWebview.js

```diff
@@ -176,9 +177,34 @@ const EknWebview = new Knowledge.Class({
                 if (scheme === 'license') {
                     let license = GLib.uri_unescape_string(uri.replace('license://', ''), null);
                     uri = Endless.get_license_file(license).get_uri();
-                }
 
-                Gtk.show_uri(null, uri, Gdk.CURRENT_TIME);
+                    let license_view = new EosKnowledgePrivate.WebViewDialog({
+                        transient_for : this.get_toplevel(),
+                        home_uri      : uri,
+                    });
+
+                    // Show local files only in the license window.
+                    // Open non file:// schemes in the system browser.
+                    license_view.webview.connect('decide-policy', (view, decision, type) => {
+                        if (type !== WebKit2.PolicyDecisionType.NAVIGATION_ACTION)
+                            return false;
```

Comment:
Nitpick: I always find the false/true return values from this handler confusing, so I like to put comments such as `// No decision made, continue with default action` or `// Decision was made`. Up to you, certainly optional.